master/buildbot/reporters/generators/comment.py

```diff
@@ -0,0 +1,154 @@
+
+from .utils import BuildStatusGeneratorMixin
+
+
+@implementer(interfaces.IReportGenerator)
+class BuildCodeIssueCommentsGenerator(BuildStatusGeneratorMixin):
+
+    wanted_event_keys = [
+        ('builds', None, 'finished'),
+    ]
+
+    compare_attrs = ['formatter']
+
+    def __init__(self, diffinfo_data_name=None, tags=None, builders=None, schedulers=None,
+                 branches=None, message_formatter=None):
+        super().__init__('all', tags, builders, schedulers, branches, subject='', add_logs=False,
+                         add_patch=False)
+        self.diffinfo_data_name = diffinfo_data_name
+        self.formatter = message_formatter
+        if self.formatter is None:
+            self.formatter = MessageFormatterFunction(lambda ctx: ctx['message'], 'plain')
+
+    @defer.inlineCallbacks
+    def generate(self, master, reporter, key, build):
+        _, _, event = key
+        if event != 'finished':
+            return None
+
+        yield utils.getDetailsForBuild(master, build,
+                                       wantProperties=self.formatter.wantProperties,
+                                       wantSteps=self.formatter.wantSteps,
+                                       wantPreviousBuild=False,
+                                       wantLogs=self.formatter.wantLogs)
+
+        if not self.is_message_needed_by_props(build):
+            return None
+        if not self.is_message_needed_by_results(build):
+            return None
+
+        report = yield self._build_report(self.formatter, master, reporter, build)
+        return report
+
+    def _get_target_changes_by_path(self, diff_info):
+        hunks_by_path = {}
+        for file_info in diff_info:
+            hunks = [(hunk['ts'], hunk['tl']) for hunk in file_info['hunks']]
+            if not hunks:
+                continue
+            hunks.sort()
+
+            hunks_by_path[file_info['target_file']] = hunks
+        return hunks_by_path
+
+    def _is_result_usable(self, result):
+        return result.get('test_code_path') is not None and result.get('line') is not None
+
+    def _is_result_in_target_diff(self, target_changes_by_path, result):
+        changes_in_path = target_changes_by_path.get(result['test_code_path'])
+        if not changes_in_path:
+            return False
+
+        result_line = result['line']
+
+        preceding_change_i = bisect.bisect(changes_in_path, (result_line + 1, -1))
+        if preceding_change_i == 0:
+            return False  # there's no change with start position earlier than the test result line
+        preceding_change_start, preceding_change_length = changes_in_path[preceding_change_i - 1]
+
+        return result_line >= preceding_change_start and \
+            result_line < preceding_change_start + preceding_change_length
+
+    @defer.inlineCallbacks
+    def _get_usable_results_in_changed_lines(self, master, target_changes_by_path, result_sets):
```

Comment:
an error could be in an untouched line (removal of used import for example).

gitlab allows you to comment on untouched lines. 

So for this generic code, I believe we shouldn't filterout

maybe we could split two sets of results something like:
```
touched_line_issues
untouched_lines_issues
```